### PR TITLE
Swap CUDA 10.1 and CPU CI for windows

### PR DESCRIPTION
**Pull request by janeyx99**
This change temporarily disables CUDA testing on PRs, but keeps it on master.
This is likely to increase the number of reverts, but this is necessary as a stop-gap measure to cap the CI costs growth.

### CODE REVIEW

File: .circleci/cimodel/data/windows_build_definitions.py [hunk 1]
```diff
@@ -138,19 +138,19 @@ def render(self):
...
     WindowsJob(1, _VC2019, CudaVersion(11, 1), master_only=True),
     WindowsJob(2, _VC2019, CudaVersion(11, 1), master_only=True),
     # VS2019 CPU-only
     WindowsJob(None, _VC2019, None),
-    WindowsJob(1, _VC2019, None, master_only=True),
-    WindowsJob(2, _VC2019, None, master_only=True),
-    WindowsJob(1, _VC2019, CudaVersion(10, 1), force_on_cpu=True, master_only=True),
+    WindowsJob(1, _VC2019, None),
+    WindowsJob(2, _VC2019, None),
+    WindowsJob(1, _VC2019, CudaVersion(10, 1), force_on_cpu=True),
```
Comment (malfet) [inline .circleci/cimodel/data/windows_build_definitions.py:153]: Are shard1 and shard2 for Windows CUDA and CPU jobs identical? If so, than perhaps we can run force_on_cpu shard1 + CPU shard2 on PR? 
```suggestion
    WindowsJob(1, _VC2019, None, master_only=True),
    WindowsJob(2, _VC2019, None),
    WindowsJob(1, _VC2019, CudaVersion(10, 1), force_on_cpu=True),
```

**Review by malfet — APPROVED** [review]

**Review by seemethere — APPROVED** [review]

**Comment by ngimel** [comment]
Hm, given how many cuda-related PRs and especially builds are broken on windows only, I'm afraid it will increase churn a lot.

**Comment by zasdfgbnm** [comment]
Does this mean no CI will be running for CUDA+Windows for PR? I feel worried about it as in the past Windows CI has been useful. (I saw Windows only failures ~twice last month)

**Comment by ptrblck** [comment]
> This change temporarily disables CUDA testing on PRs, but keeps it on master.

If I understand it correctly, Windows CI would only be triggered after the PR was merged into the master branch?
How temporary would it be, as this could indeed increase the churn of reverting faulty PRs a lot, as already mentioned, and could additionally make fixes to PRs hard (they won't be tested before the re-merge of the "fixed PR" in CI).

**Comment by seemethere** [comment]
I think based on our current cost for running Windows gpu executors it's just not tenable to run Windows CUDA tests on every PR. 

After this PR lands I'll do a follow up PR to add a CI label that we can use to trigger windows CUDA testing on PRs if requested.

It'll be similar to https://github.com/pytorch/pytorch/pull/54018

**Comment by imaginary-person** [comment]
If failures on Windows CUDA builds do not usually result from the fact that Windows CUDA CI checks use a Tesla T4 GPU (SM_75), whereas Linux CUDA CI checks use a SM_52 GPU, how about also having a Windows CI check with an SM_52 GPU, if an executor with an SM_52 GPU is significantly cheaper than an executor with an SM_75 GPU?

**Pull request closed without merging**